src/progress_bar.rs

```diff
@@ -506,10 +531,103 @@ impl WeakProgressBar {
     pub fn upgrade(&self) -> Option<ProgressBar> {
         let state = self.state.upgrade()?;
         let pos = self.pos.upgrade()?;
-        Some(ProgressBar { state, pos })
+        let ticker = self.ticker.upgrade()?;
+        Some(ProgressBar { state, pos, ticker })
+    }
+}
+
+pub(crate) struct Ticker {
+    stopping_flag: Arc<Mutex<bool>>,
```

Comment:
So what value does the `_flag` suffix add?

Reply:
Before I added the `_flag` suffix, I had `stopping` and `stopping_set` for the `Mutex` and `CondVar`, respectively. I thought `stopping_set` sounded ambiguous - i.e. does it have semantics of `HashSet` or something? So I added `_flag` to clarify. 

I guess if we merge the `Mutex<bool>` and `Condvar` into a single `Arc` we won't have to worry about naming though :)